extra/innochecksum.cc

```diff
@@ -544,24 +546,20 @@ static bool is_page_corrupted(byte *buf, bool is_encrypted, uint32_t flags)
 	return(is_corrupted);
 }
 
-/********************************************//*
- Check if page is doublewrite buffer or not.
- @param [in] page	buffer page
-
- @retval true  if page is doublewrite buffer otherwise false.
-*/
-static
-bool
-is_page_doublewritebuffer(
-	const byte*	page)
+/** Check if page is doublewrite buffer or not.
+@retval true  if page is doublewrite buffer otherwise false. */
+static bool is_page_doublewritebuffer()
 {
-	if ((cur_page_num >= extent_size)
-		&& (cur_page_num < extent_size * 3)) {
-		/* page is doublewrite buffer. */
-		return (true);
-	}
+  ut_ad(cur_space == 0);
+  /* dblwr_1, dblwr_2 wasn't initialized yet */
+  if (cur_page_num <= 5 && dblwr_1 == 0 && dblwr_2 == 0) {
+	return false;
+  }
 
-	return (false);
+  return ((cur_page_num >= dblwr_1 &&
+           cur_page_num < dblwr_1 + extent_size) ||
+          (cur_page_num >= dblwr_2 &&
+           cur_page_num < dblwr_2 + extent_size));
```

Comment:
The `if` conditions on `dblwr_1` and `dblwr_2` are redundant here. I think that we could implement an ever higher limit, simply prefix the `return` expression with `cur_page_num > FSP_DICT_HDR_PAGE_NO &&`.
```c++
  const uint32_t extent{cur_page_num & ~(extent_size - 1)};
  return extent && (extent == dblwr_1 || extent == dblwr_2);
```